plugins/rust-openid-token-plugin/src/main.rs

```diff
@@ -0,0 +1,261 @@
+use async_mutex::Mutex;
+use clap::error::{Error, ErrorKind};
+use clap::Parser;
+use reqwest::header::{HeaderName, HeaderValue};
+use reqwest::{header::CONTENT_TYPE, Client, StatusCode};
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use tonic::{transport::Server, Response, Status};
+use url::Url;
+#[macro_use]
+extern crate serde_derive;
+
+/// Requests an access token from an OpenID server and then injects the access
+/// token as an header into the API being tested by mapi.
+/// See https://mayhem4api.forallsecure.com/docs/rewrite.html for more details
+/// about rewrite plugins
+#[derive(Parser, Clone)]
+pub struct Args {
+    /// The GRPC port that will be listened on for rewrite requests from the `mapi` program.
+    /// Specify 0 to have an open port auto-selected.
+    /// May be specified using environment variable MAPI_PLUGIN_PORT.
+    #[arg(long, default_value_t = 50051, env)]
+    mapi_plugin_port: u16,
+
+    /// The URL of the openid-connect `/token` endpoint from which access tokens
+    /// will be fetched
+    #[arg(long, env)]
+    oauth_token_url: Url,
+
+    /// This option provides the body of the POST request sent to the `/token`
+    /// endpoint in order to fetch an access token. The data will be urlencoded.
+    /// Multiple calls will be joined with the `&` symbol and the full body of
+    /// the HTTP request will be sent with a `content-type` of
+    /// `application/x-www-form-urlencoded` This switch is a minimalist copy of
+    /// curl's `switch of the same name.
+    #[arg(long = "data-urlencode", env, value_parser(try_parse_data_urlencode))]
+    oath_data_urlencodes: Vec<String>,
```

Comment:
```suggestion
    oauth_data_urlencodes: Vec<String>,
```

Note -- accepting as is will break usage!

Reply:
fixed. thanks.